File: algorithms/459. Repeated Substring Pattern/solution.py
class Solution(object):
    def repeatedSubstringPattern(self, s):
        """
        :type s: str
        :rtype: bool
        """
        # refer: https://discuss.leetcode.com/topic/68206/easy-python-solution-with-explaination
        return (s +s ).find(s, 1, 2*len(s) - 1) != -1

# 107 / 107 test cases passed.
# Status: Accepted
# Runtime: 59 ms
# beats 74.48%

File: algorithms/459. Repeated Substring Pattern/solution.cc
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        string ss = s + s;
        string t(ss.begin() + 1, ss.end() - 1);
        return t.find(s) != string::npos;
    }
};

// 107 / 107 test cases passed.
// Status: Accepted
// Runtime: 32 ms
// beats 79.23%
